Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.java

```diff
@@ -128,50 +128,41 @@ public boolean onTouch(View v, MotionEvent event)
     }
 
     int pointerIndex = event.getActionIndex();
+    int actionType = event.getAction() & MotionEvent.ACTION_MASK;
+    boolean isProcessed = false;
 
-    for (InputOverlayDrawableButton button : overlayButtons)
+    // Determine the button state to apply based on the MotionEvent action flag.
+    switch (actionType)
     {
-      // Determine the button state to apply based on the MotionEvent action flag.
-      switch (event.getAction() & MotionEvent.ACTION_MASK)
-      {
-        case MotionEvent.ACTION_DOWN:
-        case MotionEvent.ACTION_POINTER_DOWN:
-          // If a pointer enters the bounds of a button, press that button.
+      case MotionEvent.ACTION_DOWN:
+      case MotionEvent.ACTION_POINTER_DOWN:
+        // If a pointer enters the bounds of a button, press that button.
+        for (InputOverlayDrawableButton button : overlayButtons)
+        {
           if (button.getBounds()
-                  .contains((int) event.getX(pointerIndex), (int) event.getY(pointerIndex)))
+            .contains((int) event.getX(pointerIndex), (int) event.getY(pointerIndex)))
           {
             button.setPressedState(true);
             button.setTrackId(event.getPointerId(pointerIndex));
             NativeLibrary.onGamePadEvent(NativeLibrary.TouchScreenDevice, button.getId(),
-                    ButtonState.PRESSED);
+              ButtonState.PRESSED);
+            isProcessed = true;
+            break;
           }
+        }
+        if(isProcessed)
           break;
-        case MotionEvent.ACTION_UP:
-        case MotionEvent.ACTION_POINTER_UP:
-          // If a pointer ends, release the button it was pressing.
-          if (button.getTrackId() == event.getPointerId(pointerIndex))
-          {
-            button.setPressedState(false);
-            NativeLibrary.onGamePadEvent(NativeLibrary.TouchScreenDevice, button.getId(),
-                    ButtonState.RELEASED);
-          }
-          break;
-      }
-    }
-
-    for (InputOverlayDrawableDpad dpad : overlayDpads)
-    {
-      // Determine the button state to apply based on the MotionEvent action flag.
-      switch (event.getAction() & MotionEvent.ACTION_MASK)
-      {
-        case MotionEvent.ACTION_DOWN:
-        case MotionEvent.ACTION_POINTER_DOWN:
-        case MotionEvent.ACTION_MOVE:
+      case MotionEvent.ACTION_MOVE:
+        for (InputOverlayDrawableDpad dpad : overlayDpads)
+        {
+          if(actionType == MotionEvent.ACTION_MOVE &&
+            dpad.getTrackId() != event.getPointerId(pointerIndex))
```

Comment:
This may be the issue here? I think the dpad trackId needs to be set on the Down action if we now want to check against it. Here is what I did in the touch pointer PR.

https://github.com/dolphin-emu/dolphin/blob/808d3c0e313f8f5614e2314aa61d0450862934c9/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.java#L176